test/startservers.py

```diff
@@ -10,6 +11,132 @@
+        ('boulder-va-1', 'boulder-va-2')),
+    Service('boulder-ca-b',
+        8104,
+        ('./bin/boulder-ca', '--config', os.path.join(config_dir, 'ca-b.json'), '--ca-addr', 'ca2.boulder:9093', '--ocsp-addr', 'ca2.boulder:9096', '--debug-addr', ':8101'),
+        ('boulder-va-1', 'boulder-va-2')),
+    Service('akamai-test-srv',
+        6789,
+        ('./bin/akamai-test-srv', '--listen', 'localhost:6789', '--secret', 'its-a-secret'),
+        ('boulder-ca-a', 'boulder-ca-b')),
+    Service('akamai-purger',
+        9666,
+        ('./bin/akamai-purger', '--config', os.path.join(config_dir, 'akamai-purger.json')),
+        ('akamai-test-srv',)),
+    Service('ocsp-updater',
+        8006,
+        ('./bin/ocsp-updater', '--config', os.path.join(config_dir, 'ocsp-updater.json')),
+        ('akamai-purger',)),
+    Service('boulder-ra-1',
+        8002,
+        ('./bin/boulder-ra', '--config', os.path.join(config_dir, 'ra.json'), '--addr', 'ra1.boulder:9094', '--debug-addr', ':8002'),
+        ('ocsp-updater',)),
+    Service('boulder-ra-2',
+        8102,
+        ('./bin/boulder-ra', '--config', os.path.join(config_dir, 'ra.json'), '--addr', 'ra2.boulder:9094', '--debug-addr', ':8102'),
+        ('ocsp-updater',)),
+    Service('nonce-service-taro',
+        8111,
+        ('./bin/nonce-service', '--config', os.path.join(config_dir, 'nonce.json'), '--addr', 'nonce1.boulder:9101', '--debug-addr', ':8111', '--prefix', 'taro'),
+        ('boulder-ra-1', 'boulder-ra-2')),
+    Service('nonce-service-zinc',
+        8112,
+        ('./bin/nonce-service', '--config', os.path.join(config_dir, 'nonce.json'), '--addr', 'nonce2.boulder:9101', '--debug-addr', ':8112', '--prefix', 'zinc'),
+        ('boulder-ra-1', 'boulder-ra-2')),
+    Service('boulder-wfe2',
+        4001,
+        ('./bin/boulder-wfe2', '--config', os.path.join(config_dir, 'wfe2.json')),
+        ('nonce-service-taro', 'nonce-service-zinc')),
+    Service('boulder-wfe',
+        4000,
+        ('./bin/boulder-wfe', '--config', os.path.join(config_dir, 'wfe.json')),
+        ('nonce-service-taro', 'nonce-service-zinc')),
+    Service('log-validator',
+        8016,
+        ('./bin/log-validator', '--config', os.path.join(config_dir, 'log-validator.json')),
+        ('boulder-wfe2', 'boulder-wfe')),
+)
+
+def _service_toposort(services):
+    """Yields Service objects in topologically sorted order.
+
+    No service will be yielded until every service listed in its deps value
+    has been yielded.
+    """
+    ready = set([s for s in services if not s.deps])
+    blocked = set(services) - ready
+    done = set()
+    while ready:
+        service = ready.pop()
+        yield service
+        done.add(service.name)
+        new = set([s for s in blocked if all([d in done for d in s.deps])])
+        ready |= new
+        blocked -= new
+    if blocked:
+        print("WARNING: services with unsatisfied dependencies:")
```

Comment:
This should be a fatal error in my opinion. We expect all the services to be able to come up.

Reply:
Added an exception